amoro-ams/src/main/java/org/apache/amoro/server/persistence/SqlSessionFactoryProvider.java

```diff
@@ -133,6 +148,12 @@ public void init(Configurations config) throws SQLException {
     createTablesIfNeed(config);
   }
 
+  private void registerDialectAliases() {
+    registerDialectAlias("postgresql", PostgreSqlDialect.class);
+    registerDialectAlias("mysql", MySqlDialect.class);
+    registerDialectAlias("derby", SqlServerDialect.class);
```

Comment:
According to https://github.com/pagehelper/Mybatis-PageHelper
We may use `SqlServer2012Dialect` for derby.

Reply:
fixed, and add some ut to cover it